File: src/main/java/ir/mimdolt/store/business/filemanager/FileManagerService.java
package ir.mimdolt.store.business.filemanager;



import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.HashMap;

/**
 * Created by Hadi Jeddi Zahed on 11/9/2016.
 */
public interface FileManagerService {


    HashMap<String, Object> createFolder(HashMap<String, String> params) throws Exception;

    HashMap<String, Object> renameUrl(HashMap<String, String> params) throws Exception;

    void remove(String path) throws Exception;

    void upload(HttpServletRequest request, HttpServletResponse response) throws Exception;

    byte[] download(String action, String path)throws Exception;
}

File: src/main/webapp/mimdolt/pages/back/filemanager/fileManagerService.js
/**
 * Created by Hadi Jeddi Zahed on 11/13/2016.
 */
angular.module('myApp').factory("fileManagerService", [
    "$resource",
    function ($resource) {
        return $resource("/api/admin/fm", {}, {
            query: {method: "GET", isArray: true},
            create: {method: "POST"},
            remove: {method: "DELETE", url: "/api/admin/fm/delete/:path",params:{
                path:'@path'
            }},
            get: {
                method: "GET", url: "/api/fm/download/:path",
                params:{
                    path:'@path'
                }
            }
        });
    }
]);